app/controllers/journalist/articles_controller.rb

```diff
@@ -0,0 +1,46 @@
+class Journalist::ArticlesController < ApplicationController
+    before_action :check_journalist
+    before_action :authenticate_user!
+    def index
+        @articles = Article.all.where(user_id: current_user.id)
+    end
+
+    def show
+        @article = Article.find(params[:id])
+    end
+    
+    def edit
+        @article = Article.find(params[:id])
+    end
+
+    def update
+        @article = Article.find(params[:id])
+       
+        if @article.update(article_params)
+            redirect_to journalist_article_path, notice: 'Article was successfully updated.'
+        else
+            redirect_to @article, notice: 'Article was not updated.'
+        end
+    end
+
+    def destroy
```

Comment:
Set a conditional that the user(journalist) can only access this article if he is the owner.

Reply:
done